### DISCUSSION TITLE
How to put data dependent constraints on GP?

### DISCUSSION THREAD

**Discussion by pranav997** [cornellius-gp/gpytorch#2298] [Q&A] [answered]
Hi,
Thank you for this wonderful library. I was wondering if there is a way to put data dependent constraints on function. I have a four dimensional function x=[x1,x2,x3,x4]. I would want to put constraints such as f(x2)>f(x'2) for x2>x'2. I also want to put bound constraints which are data dependent such as f(x1)<x1 for different values of x1. Is there any way to do that?

I found [libraries](https://github.com/brunzema/constrained_gps) that can put linear constraints on derivatives but constraints are quite general. Any way to make them data dependent?

**Comment by gpleiss** [comment, marked as answer]
At the moment, there is no good way to do this in GPyTorch.
